### PR TITLE
Handle unknown assets in Uploads

### PR DESCRIPTION
**Pull request by mikkio-j**
Issue: #532

### CODE REVIEW

File: src/hooks/useUploadsManager/useUploadsManager.tsx [hunk 1]
```diff
@@ -47,7 +57,7 @@ export const UploadManagerProvider: React.FC = ({ children }) => {
   const [assetsFiles, setAssetsFiles] = useState<AssetFile[]>([])
   const { activeChannelId } = useUser()
   const { loading: channelLoading } = useChannel(activeChannelId ?? '')
-  const { loading: videosLoading } = useVideos(
+  const { loading: videosLoading, client } = useVideos(
```
Comment (kdembler) [inline src/hooks/useUploadsManager/useUploadsManager.tsx]: Why do we still need that fetch?

File: src/hooks/useUploadsManager/useUploadsManager.tsx [hunk 2]
```diff
@@ -69,34 +81,154 @@ export const UploadManagerProvider: React.FC = ({ children }) => {
...
+
+        const isMediaPresentInUploads = uploadsState.some(
+          (asset) => asset.contentId === video.mediaDataObject?.joystreamContentId
+        )
+        const isThumbnailPresentInUploads = uploadsState.some(
+          (asset) => asset.contentId === video.thumbnailPhotoDataObject?.joystreamContentId
+        )
+
+        if (isMediaPending && !isMediaPresentInUploads) {
+          addAsset({
```
Comment (kdembler) [inline src/hooks/useUploadsManager/useUploadsManager.tsx]: I'm wondering, what's the point of adding that all to the local storage?

The approach we take here currently is that we use local storage as the source of truth - that's what we will be using to display the data later on. And we do a sync at the beginning to make it updated. But what if we took a different approach and decided that the data from the query node is the source of truth and whatever we have in the local storage can be an addition to that?

Comment (kdembler) [reply src/hooks/useUploadsManager/useUploadsManager.tsx]: Does that make sense? I think in that case we wouldn't need to add those assets to the state at all

File: src/hooks/useUploadsManager/useUploadsManager.tsx [hunk 3]
```diff
@@ -22,62 +31,195 @@ export const UploadManagerProvider: React.FC = ({ children }) => {
...
-  const { loading: channelLoading } = useChannel(activeChannelId ?? '')
-  const { loading: videosLoading } = useVideos(
-    {
-      where: {
-        id_in: uploadsStateIds,
-      },
-    },
-    { skip: !uploadsStateIds.length }
-  )
+  const [queryNodePendingAssets, setQueryNodePendingAssets] = useState<AssetUpload[]>([])
```
Comment (kdembler) [inline src/hooks/useUploadsManager/useUploadsManager.tsx]: Why not make this part of the store?

Comment (mikkio-j) [reply src/hooks/useUploadsManager/useUploadsManager.tsx]: I thought we don't want to make query node data part of the store. Can make that though - that will require some additional usage of `updateAsset` on initial sync up

Comment (kdembler) [reply src/hooks/useUploadsManager/useUploadsManager.tsx]: Ah okay I misunderstood what this is. Let me think about this

**Review by kdembler — APPROVED** [review]

**Pull request merged**